libnavigation-util/src/main/java/com/mapbox/navigation/utils/internal/LoggerFrontend.kt

```diff
@@ -24,6 +27,13 @@ internal class MapboxCommonLoggerFrontend : LoggerFrontend {
         NativeLoggerWrapper.debug(message, NAV_SDK_CATEGORY)
     }
 
+    override fun logD(category: String?, lazyMsg: () -> String) {
```

Comment:
I like that. The lambda would still be created but I don't see how we could evade that.
Even if we make the logD method inline (which is not possible with interfaces), `LogConfiguration.getLoggingLevel()` is not a constant, so the lambda would not be removed anyway.

Reply:
Made methods inline to avoid creation of an object for lambda